### PR TITLE
[feat] cross compile

### PR DESCRIPTION
**Pull request by denyncrawford**
Adds cross-compiling

### CODE REVIEW

**Comment by denyncrawford** [comment]
Waiting for tests.

**Comment by ramirobelmonte** [comment]
Falló la prueba en GNU/Linux

image

**Comment by Dave136** [comment]
I ran the app with Ubuntu Budgie 20.04:
Screenshot from 2022-03-19 18:33:10

Result:
Screenshot from 2022-03-19 18:33:24

File: modules/astrodon-build/mod.ts [hunk 1]
```diff
@@ -64,7 +61,13 @@ export class Builder {
     Deno.mkdir(this.config.dist, { recursive: true });
 
     // Preatere the final executable
-    const final_bin_path = `${join(this.config.dist, this.info.name)}${Deno.build.os === "windows" ? ".exe" : ""}`;
+
+    const final_bin_path = `${join(this.config.dist, this.info.name)}${
+      ["darwin", "linux"].includes(this.os)
+        ? `_${this.config?.build?.targets?.[this.os]?.sufix ?? this.os}`
```
Comment (clemens-tolboom) [inline modules/astrodon-build/mod.ts:67]: Nitpicking it is suffix

File: modules/astrodon-build/mod.ts [hunk 2]
```diff
@@ -97,7 +100,13 @@ export class Builder {
    * Create an installer for the compiled executable
    */
   async makeInstaller() {
-    const src_path = `${join(this.config.dist, this.info.name)}${Deno.build.os === "windows" ? ".exe" : ""}`;
+    
+    const src_path = `${join(this.config.dist, this.info.name)}${
+      ["darwin", "linux"].includes(this.os)
+        ? `_${this.config?.build?.targets?.[this.os]?.sufix ?? this.os}`
```
Comment (clemens-tolboom) [inline modules/astrodon-build/mod.ts:106]: suffix

File: modules/astrodon/mod.ts [hunk 3]
```diff
@@ -12,8 +14,17 @@ export interface AppInfo {
   resources: string[];
 }
 
+export interface AppBuildOptions {
+  targets?: {
+    [key in OSNames]?: {
+      sufix?: string;
```
Comment (clemens-tolboom) [inline modules/astrodon/mod.ts:20]: again

File: modules/astrodon-build/mod.ts [hunk 4]
```diff
@@ -64,7 +61,13 @@ export class Builder {
     Deno.mkdir(this.config.dist, { recursive: true });
 
     // Preatere the final executable
```
Comment (clemens-tolboom) [inline modules/astrodon-build/mod.ts:63]: >     // Preatere the final executable
Side note: what does this mean?

File: modules/astrodon-build/mod.ts [hunk 5]
```diff
@@ -64,7 +61,13 @@ export class Builder {
...
-    const final_bin_path = `${join(this.config.dist, this.info.name)}${Deno.build.os === "windows" ? ".exe" : ""}`;
+
+    const final_bin_path = `${join(this.config.dist, this.info.name)}${
```
Comment (clemens-tolboom) [inline modules/astrodon-build/mod.ts:65]: This looks way to magic too me. Maybe rewrite oradding a comment could help?

It seems weird too compared to the global README:

> Only Windows and Linux are supported

Comment (marc2332) [reply modules/astrodon-build/mod.ts]: It should be be more clear now. Btw, you are reading from the readme on the main branch, this PR is targeted to `feature/deno_tauri` which is a pretty big rewrite and it supports MacOS too.

Thanks for pointing out :))

File: modules/astrodon-build/mod.ts (hunk 4 above)
Comment (marc2332) [reply modules/astrodon-build/mod.ts]: It was a typo, it mean "Prepare"

File: modules/astrodon-build/mod.ts (hunk 1 above)
Comment (marc2332) [reply modules/astrodon-build/mod.ts]: Nice catch!

File: modules/astrodon-build/mod.ts (hunk 5 above)
Comment (clemens-tolboom) [reply modules/astrodon-build/mod.ts]: Oeps #26 ... should I close?

**Pull request merged**